openedx/features/course_experience/views/course_home.py

```diff
@@ -248,3 +264,26 @@ def render_to_fragment(self, request, course_id=None, **kwargs):
         }
         html = render_to_string('course_experience/course-home-fragment.html', context)
         return Fragment(html)
+
+
+def _get_next_up_banner_fragment(assignment_title, resume_course_url):
+    if assignment_title and resume_course_url:
+        return _generate_up_next_banner_html(assignment_title, resume_course_url)
+    return None
+
+
+def _generate_up_next_banner_html(assignment_title, resume_course_url, assignment_duration='10 min'):
+    next_up_message = _(
+        u"{div_open}<b>Get started on what's next:</b> {a_open}"
+        u"{assignment_title}{a_close} ({assignment_duration}){div_close}"
+    ).format(
+        div_open=HTML('<div class="up-next-banner">'),
+        a_open=HTML('<a href="{resume_course_url}">').format(
+            resume_course_url=resume_course_url,
+        ),
+        assignment_title=assignment_title,
+        a_close=HTML('</a>'),
+        assignment_duration=assignment_duration,
+        div_close=HTML('</div>'),
+    )
+    return Fragment(HTML(next_up_message))
```

Comment:
Any reason you didn't just make this a template with the HTML, rather than smashing it together in python?

Reply:
Mostly to make it consistent with the other banners that get rendered in this view, it _did_ feel gross to write that fragment though honestly, though felt keeping it the same was more readable overall. https://github.com/edx/edx-platform/blob/master/openedx/features/course_experience/templates/course_experience/course-home-fragment.html#L65-L73

Reply:
I was thinking about https://github.com/edx/edx-platform/blob/2ad06c5f306beaf2a86792005ddeba830de2501a/openedx/features/course_experience/views/course_home.py#L275-L289 in particular. Basically, that could be done by putting most of that code into a template, which would handle the escaping for your much more cleanly, and I think the end result would be clearer.

Reply:
I get that a template is probably cleaner, but with the offer_banner_fragment we do something almost exactly like what is done here, https://github.com/edx/edx-platform/blob/2ad06c5f306beaf2a86792005ddeba830de2501a/openedx/features/discounts/utils.py#L122-L145 should I refactor that too? I'd rather consistency for cleanliness over pure readability.

Reply:
I honestly think that those banners are fare enough apart (different django apps) that we don't need to enforce an artificial consistency between them. I'd love to see the other one refactored too, but that code is owned by the revenue squad.